docs/advanced.md

```diff
@@ -111,3 +111,48 @@ password example-password
 
 ...
 ```
+
+## HTTP Proxying
+
+HTTPX supports setting up proxies the same way that Requests does via the `proxies` parameter.
+
+```python
+>>> client = httpx.Client(proxies={
+  "http": "http://127.0.0.1:3080",
+  "https": "http://127.0.0.1:3080"
+})
+```
+
+Proxies can be configured per scheme, scheme+host, for all schemes of a host, or for all requests:
+
+```python
+>>> client = httpx.Client(proxies={
+    "http://example.com":  "...",  # Host+Scheme
+    "all://example.com":  "...",  # Host
+    "http": "...",  # Scheme
+    "all": "...",  # All
+})
+>>> client = httpx.Client(proxies="...")  # Shortcut for 'all'
+```
+
+!!! warning
+    It is recommended for privacy reasons to use HTTPS and tunnel requests if possible.
+    Proxies can read your traffic even if the `proxy_url` parameter uses HTTPS
+    unless you make requests to the target domain with HTTPS through a `CONNECT` TCP tunnel.
```

Comment:
It seems to me both sentences here convey the same tl;dr. So, may I suggest…?

> To make sure that proxies cannot read your traffic, and even if the `proxy_url` uses HTTPS, it is recommended to use HTTPS and tunnel requests if possible.

Also, I'm now wondering what prevents us from applying this recommendation by default? Does it depend on proxies capabilities?

Reply:
Many proxies (especially public/free) don't offer HTTPS for both hops unfortunately.
Users will have to determine the capabilities of their proxy and hopefully configure properly, this warning will hopefully encourage people to try HTTPS first?